### PR TITLE
fig 3 fix

### PR DESCRIPTION
**Pull request by Farnazmdi**
I have been trying to fit censored data when the model doesn't know the data is censored (to create the suggested part (c)) but it fails to run. There is an assertion error for estimating the transition probability matrix. 
So, instead of keeping the censored observations and treating them like observed observations, I marked them as 'nan' so it seems we removed the censored observations. 
Do you expect it to be this bad? @aarmey 

image

### CODE REVIEW

**Comment by aarmey** [comment]
But this doesn't match the process we're trying to imitate.

What is the assertion error? You should fix that.

**Comment by Farnazmdi** [comment]
By "uncensored model" you mean the model thinks of censored data as uncensored, right? 

Sometimes the `res.success` from the gamma_estimator fails, sometimes `np.all(np.isfinite(T_estimate))` from M_T step fails.

**Comment by aarmey** [comment]
Censored is defined by the Bernoulli being NaN, right?

If you print out `res` when it fails that check, what does it show?

**Comment by aarmey** [comment]
You might have to mark all the censored cells at the end of the experiment with a non-NaN Bernoulli.

**Comment by Farnazmdi** [comment]
I was focusing on the 4th and 5th elements of the `cell.obs`. If cell.obs[4] == 0 it means G1 is censored, if == 1 means uncensored, and if 'nan' means unobserved. The 5th element is the same for G2 phase.

**Comment by aarmey** [comment]
In the "broken" case, you want the model to think these are fully observed cells, even though they aren't. So, they should be marked with 1 as uncensored.

**Comment by Farnazmdi** [comment]
I was marking cell.obs[4] == cell.obs[5] == 1 so that the model thinks they are uncensored. But never got a successful run.

**Comment by Farnazmdi** [comment]
This is the message when the convergence is not successful:
image

**Comment by aarmey** [comment]
Ok, you can comment out that convergence assert. Something weird is going on there, but it’s close to the solution.

**Comment by Farnazmdi** [comment]
The problem with other assertion errors originates from pdf calculation for some of the cells.
When a pdf is 'nan', how is it different from pdf being 0?

**Comment by aarmey** [comment]
Hmm... do you know the inputs that result in NaN? It shouldn't happen, and is not the same as 0.

https://en.wikipedia.org/wiki/IEEE_754-1985

**Comment by Farnazmdi** [comment]
There are many cells. I will try to print out their observations tomorrow.

**Comment by Farnazmdi** [comment]
image
I had to leave the unobserved cells to be marked as `nan`. I think it is fixed now.

File: lineage/states/stateCommon.py [hunk 1]
```diff
@@ -68,15 +68,14 @@ def gamma_estimator(gamma_obs, time_cen, gammas, x0):
     assert gammas.shape[0] == gamma_obs.shape[0]
     arrgs = (gamma_obs[time_cen == 1], gammas[time_cen == 1], gamma_obs[time_cen == 0], gammas[time_cen == 0])
     opt = {'gtol': 1e-12, 'ftol': 1e-12}
-    bnd = (1.0, 800.0)
+    bnd = (1.0, 100.0)
```
Comment (aarmey) [inline lineage/states/stateCommon.py:71]: I assume this bound is still sufficiently wide to cover all feasible values?

**Review by aarmey — APPROVED** [review]
One question—depending upon the answer there this is ready.

File: lineage/states/stateCommon.py (hunk 1 above)
Comment (Farnazmdi) [reply lineage/states/stateCommon.py:71]: Oh I tried this to fix the error in the tests. I was passing almost 99% when I changed it to 100. I'm not sure if this was the problem though. Should I bring that back?

Comment (Farnazmdi) [reply lineage/states/stateCommon.py:71]: To answer your question, I think it is.

Comment (aarmey) [reply lineage/states/stateCommon.py:71]: Ok, this is good to merge.

**Pull request merged**